### PR TITLE
MRG, FIX: Refactor transformations

### PR DESCRIPTION
**Pull request by larsoner**
This is designed to simplify our use of coordinate transforms so that the code is more maintainable and understandable, especially when it comes to volumetric work.

Locally I have one failing test:
```
- mne/tests/test_morph.py:523 test_volume_labels_morph
```
I suspect it's just a numerical precision thing changing the voxel counts in `mri_resolution=True`. We'll see if some of the CIs pass.

This PR was not meant to change functionality, but I realized we should do two things:

1. Deprecate the `mode` arg which was used to switch between `nibabel` and `freesurfer` for reading MGH MRI headers. At some point we added our own reader for MGH headers, so there is no reason to have this argument anymore. I'm tempted to remove our internal reader entirely since `nibabel` is easy enough to install via `pip` and it's required for almost all volumetric stuff (where it's needed in our code) nowadays anyway.
2. Fix the `plot_mri_dipole_locations` functionality when `coord_frame='head'`

Also gets rid of the `_tkr` checks because that was part of nibabel 2.0 and we require 2.1+.

Todo:

- [x] ~Simplify~ Fix `plot_dipole_locations` orthoview mode head coord frame
- [x] Fix failing tests

### CODE REVIEW

**Comment by larsoner** [comment]
Okay it looks like plot_dipole_locations is broken in `master`. Here is `examples/inverse/plot_mixed_norm_inverse.py` in MRI coords:

Figure_2_mri_before

If you look at the head coord frame, it's actually wrong -- the location jumps the Sylvian fissure:

Figure_2_head_before

With the changes in this PR, it's fixed:

Figure_2_head_after

**Comment by larsoner** [comment]
Okay this is ready for review/merge from my end

File: examples/inverse/plot_mixed_norm_inverse.py [hunk 1]
```diff
@@ -80,13 +70,13 @@
 idx = np.argmax([np.max(np.abs(dip.amplitude)) for dip in dipoles])
 plot_dipole_locations(dipoles[idx], forward['mri_head_t'], 'sample',
                       subjects_dir=subjects_dir, mode='orthoview',
-                      idx='amplitude')
+                      idx='amplitude', coord_frame='head')
```
Comment (agramfort) [inline examples/inverse/plot_mixed_norm_inverse.py]: why do you plot in head? it makes the MRI clipped and it's harder to see where the dipole is with a tilted brain

Comment (larsoner) [reply examples/inverse/plot_mixed_norm_inverse.py]: Mostly just to see it work properly, will revert

**Pull request merged**